src/timeMachine/utils/rawFluxDataTable.ts

```diff
@@ -14,6 +15,165 @@ export interface ParseFilesResult {
+  const {
+    fluxGroupKeyUnion,
+    table,
+    table: {columnKeys: keys},
+  } = fromFlux(response)
+
+  const columnKeys = [
+    'result',
+    'table',
+    ...keys.filter(k => k !== 'result' && k !== 'table'),
+  ]
+
+  const tables = table.getColumn('table')
+  const values = table.getColumn('result')
+  let [currVal] = values
+  let [currTable] = tables
+  let columnHeaders = []
+  const tableData = []
+  // find where the table splits based on yielded result names
+  const groupSet = new Set(fluxGroupKeyUnion)
+  for (let i = 0; i < tables.length; i++) {
+    if (i === 0) {
+      // build out the columnHeader = chunk[3]
+      columnHeaders = columnKeys.filter(col =>
+        filterColumnHeaders(col, table, i)
+      )
+      const {groupHeaders, dataTypes, defaultRow} = getGroupDataDefaultHeaders(
+        columnHeaders,
+        table,
+        groupSet,
+        values[i]
+      )
+      tableData.push(groupHeaders, dataTypes, defaultRow, [
+        '',
+        ...columnHeaders,
+      ])
+    }
+    if (values[i] !== currVal || tables[i] !== currTable) {
+      // sets the boundaries for the chunk based on different yields or tables
+      currVal = values[i]
+      currTable = tables[i]
+      // create a chunk
+      // build out the columnHeader = chunk[3]
+      columnHeaders = columnKeys.filter(col =>
+        filterColumnHeaders(col, table, i)
+      )
+      const {groupHeaders, dataTypes, defaultRow} = getGroupDataDefaultHeaders(
+        columnHeaders,
+        table,
+        groupSet,
+        values[i]
+      )
+      tableData.push([], groupHeaders, dataTypes, defaultRow, [
+        '',
+        ...columnHeaders,
+      ])
+    }
+    // construct the chunk of data for this given row
+    tableData.push(getRowData(columnHeaders, table, i))
+  }
+
+  return tableData
+}
+
+type Headers = {
+  groupHeaders: string[]
+  dataTypes: string[]
+  defaultRow: string[] // TODO(ariel): can this be a number or bool?
+}
+
+const getGroupDataDefaultHeaders = (
+  columnHeaders: string[],
+  table: Table,
+  groupSet: Set<string>,
+  value: string | number | boolean
+): Headers => {
+  // based on the columnHeader we can get the:
+  // #group, true, false ...(chunk[0])
+  const groupHeaders = buildGroupHeaders(columnHeaders, groupSet)
+  // #datatype, 'string', 'long' ...(chunk[1])
+  const dataTypes = [
+    '#datatype',
+    ...columnHeaders.map(col => table.getOriginalColumnType(col)),
+  ]
+  // #default, result, '', '' ...(chunk[2])
+  const defaultRow = ['#default', `${value}`]
+  while (defaultRow.length <= columnHeaders.length) {
+    defaultRow.push('')
+  }
+  return {
+    groupHeaders,
+    dataTypes,
+    defaultRow,
+  }
+}
+
+const filterColumnHeaders = (
+  column: string,
+  table: Table,
+  index: number
+): boolean => {
+  const columnType: any = table.getColumnType(column)
+  const values = table.getColumn(column, columnType)
+  return values[index] !== undefined
+}
+
+const buildGroupHeaders = (
```

Comment:
not a blocker, but functions don't come for free. when dealing with large datasets, you wanna minimize the amount of memory duplication in temporary allocations (like how the reduce function duplicates acc every time instead of operating on memory in place) as these dont get removed immediately (i forget the default chrome garbage collection frequency) and the number of function calls as these create similar temporary [memory allocations](https://en.wikipedia.org/wiki/Call_stack).

for loops > helper functions, but only when performance matters more than ergonomics